From 933dde66ca0fa72823e3f1d525099d2edf9d3472 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Sun, 8 Mar 2026 15:17:41 +0100
Subject: [PATCH 062/121] winepulse: Stop forcing period to twice the minimum
 size.

Since pipewire-1.6, minimum quant value has been increased to 256: no need to multiply it manually anymore.

https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/7bd65cfe9385b4e005b7f5a85c9f1bdfa8a7b6d6
---
 dlls/winepulse.drv/pulse.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
index b14cbdb763d..45cfb2f4e71 100644
--- a/dlls/winepulse.drv/pulse.c
+++ b/dlls/winepulse.drv/pulse.c
@@ -832,7 +832,6 @@ static void pulse_probe_settings(pa_mainloop *ml, pa_context *ctx, int render, c
 
     if (length) {
         REFERENCE_TIME period_to_set = pa_bytes_to_usec(10 * length, &ss);
-        int multiplier = 2; // set 2x actual minimum unless custom period was specified
         static int periodval = -1;
         if (periodval == -1) {
             const char* penv = getenv("STAGING_AUDIO_PERIOD");
@@ -842,14 +841,12 @@ static void pulse_probe_settings(pa_mainloop *ml, pa_context *ctx, int render, c
             }
         }
 
-        if (periodval > 0) {
+        if (periodval > 0)
             period_to_set = periodval;
-            multiplier = 1;
-        }
 
         /* overwrite default with minimum (for playback devices) */
         *min_period = find_closest_to_next_integer(period_to_set, ss.rate, 10000000);
-        *def_period = find_closest_to_next_integer(*min_period * multiplier, ss.rate, 10000000);
+        *def_period = find_closest_to_next_integer(*min_period, ss.rate, 10000000);
     }
 
     wfx->wFormatTag = WAVE_FORMAT_EXTENSIBLE;
-- 
2.54.0

